launch_snakemake.py

```diff
@@ -339,25 +351,65 @@ def analysis_main(args, output, runnormal=False, normalname=False, normalfastqs=
+            configfile = f"{runconfigs}/{tumorid}_config.json"
         else:
-            snakemake_args = f"snakemake -s pipeline.snakefile --configfile {runconfigs}/{normalid}_config.json --use-singularity --singularity-args '-e --bind {binddir_string} --bind /medstore --bind /seqstore --bind /apps' --cluster-config configs/cluster.yaml --cluster \"qsub -S /bin/bash -pe mpi {{cluster.threads}} -q {{cluster.queue}} -N {{cluster.name}} -o {samplelogs}/{{cluster.output}} -e {samplelogs}/{{cluster.error}} -l {{cluster.excl}}\" --jobs 999 --latency-wait 60 --directory {output} --shadow-prefix {shadow_dir} &>> {samplelog}"
-        # >>>>>>>>>>>> Start pipeline
-        subprocess.run(snakemake_args, shell=True, env=my_env) # Shellscript pipeline
-
+            configfile = f"{runconfigs}/{normalid}_config.json"
+
+        singularity_args = [
+            "-e",
+            "--bind", binddir_string,
+            "--bind", "/medstore",
+            "--bind", "/seqstore",
+            "--bind", "/apps",
+        ]
+
+        cluster_args = [
+            "qsub",
+            "-S", "/bin/bash",
+            "-pe", "mpi", "{cluster.threads}",
+            "-q", "{cluster.queue}",
+            "-N", "{cluster.name}",
+            "-o", f"{samplelogs}/{{cluster.output}}",
+            "-e", f"{samplelogs}/{{cluster.error}}",
+            "-l", "{cluster.excl}"
+        ]
+
+        # Create DAG of pipeline
+        snakemake_args_DAG = [
+            "snakemake", "-s", "pipeline.snakefile",
+            "--configfile", configfile,
+            "--dag", "|", "dot", "-Tsvg", ">", f"{samplelogs}/dag_{current_date}.svg"
+        ]
+        snakemake_args_DAG_str = " ".join(snakemake_args_DAG)
+        subprocess.run(snakemake_args_DAG_str, shell=True, env=my_env)
+
+        snakemake_args = [
+            "snakemake", "-s", "pipeline.snakefile",
+            "--configfile", configfile,
+            "--use-singularity", "--singularity-args", " ".join(singularity_args),
+            "--cluster-config", "configs/cluster.yaml",
+            "--cluster", " ".join(cluster_args),
+            "--jobs", "999",
+            "--latency-wait", "60",
+            "--directory", output,
+            "--shadow-prefix", shadow_dir
+        ] + dev_args
+
+        # Execute Snakemake command
+        subprocess.run(snakemake_args, env=my_env, check=True)
+
+    except subprocess.CalledProcessError as e:
+        print(f"Error running Snakemake: {e}")
```

Comment:
switch print to logger?

Reply:
Good idea! #[b8c0996](https://github.com/ClinicalGenomicsGBG/wgs_somatic/pull/84/commits/b8c0996b197f33ad425fc0704d5b52ef3d81f4ff)